### PR TITLE
Enhancements for spec for Exception

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: core/exception/case_compare_spec.rb [hunk 1]
```diff
@@ -1,5 +1,35 @@
 require_relative '../../spec_helper'
 
 describe "SystemCallError.===" do
-  it "needs to be reviewed for spec completeness"
+  before :all do
+    @example_errno = 22
```
Comment (eregon) [inline core/exception/case_compare_spec.rb]: Could you use `Errno::EINVAL::Errno` here? That way, we don't hardcode any number.

File: core/exception/case_compare_spec.rb [hunk 2]
```diff
@@ -1,5 +1,35 @@
...
+    @example_errno_class = Errno::EINVAL
+  end
+
```
Comment (eregon) [inline core/exception/case_compare_spec.rb]: Could you add a spec for `Errno::EINVAL.should === Errno::EINVAL.new` which is the simplest case?

File: core/exception/system_call_error_spec.rb [hunk 3]
```diff
@@ -20,41 +20,82 @@ def initialize
 end
 
 describe "SystemCallError.new" do
+  before :all do
+    # EINVAL/22 should be consistent across platforms.
+    @example_errno = 22
```
Comment (eregon) [inline core/exception/system_call_error_spec.rb]: Same as above, use `Errno::EINVAL::Errno`

File: core/exception/system_call_error_spec.rb [hunk 4]
```diff
@@ -20,41 +20,82 @@ def initialize
...
+    e = SystemCallError.new(@example_errno)
+    e.should be_kind_of(SystemCallError)
+    e.should be_an_instance_of(@example_errno_class)
   end
 
   it "accepts an optional custom message preceding the errno" do
-    exc = SystemCallError.new("custom message", 22)
-    exc.should be_an_instance_of(Errno::EINVAL)
-    exc.errno.should == 22
-    exc.message.should == "Invalid argument - custom message"
```
Comment (eregon) [inline core/exception/system_call_error_spec.rb]: I think this check was useful, and clearer about what the full message should be, could you keep it?

Comment (BurdetteLamar) [reply core/exception/system_call_error_spec.rb]: Yes, I agree, and will restore.

Earlier I had tried to select an example name/num from Errno.constants, and so had removed the specific msg part here.
I was surprised to find that Errno.constants.index(:EINVAL) is not 22 in either my Win or Linux OS!  Apparently the 22 is computed or looked up here.

File: core/exception/case_compare_spec.rb (hunk 1 above)
Comment (BurdetteLamar) [reply core/exception/case_compare_spec.rb]: Wonderful!  I tried a different strategy (see below) that didn't work.

File: core/exception/case_compare_spec.rb (hunk 2 above)
Comment (BurdetteLamar) [reply core/exception/case_compare_spec.rb]: Will do.

Comment (BurdetteLamar) [reply core/exception/case_compare_spec.rb]: What would be the description?

  it "???" do
    Errno::EINVAL.should === Errno::EINVAL.new
  end

File: core/exception/case_compare_spec.rb [hunk 5]
```diff
@@ -1,5 +1,39 @@
...
 describe "SystemCallError.===" do
-  it "needs to be reviewed for spec completeness"
+  before :all do
+    @example_errno_class = Errno::EINVAL
+    @example_errno = @example_errno_class::Errno
+  end
+
+  it "???" do
+    Errno::EINVAL.should === Errno::EINVAL.new
+  end
```
Comment (BurdetteLamar) [inline core/exception/case_compare_spec.rb]: Oops.  Meant to wait to hear about this description.

Comment (BurdetteLamar) [reply core/exception/case_compare_spec.rb]: @eregon, can you add it here?

Comment (eregon) [reply core/exception/case_compare_spec.rb]: `returns true for an instance of the same class` would be good

File: core/exception/case_compare_spec.rb [hunk 6]
```diff
@@ -1,5 +1,39 @@
 require_relative '../../spec_helper'
 
 describe "SystemCallError.===" do
-  it "needs to be reviewed for spec completeness"
+  before :all do
+    @example_errno_class = Errno::EINVAL
+    @example_errno = @example_errno_class::Errno
+  end
+
+  it "???" do
```
Comment (eregon) [inline core/exception/case_compare_spec.rb]:
```suggestion
  it "returns true for an instance of the same class" do
```

**Pull request merged**